gateway/ln-gateway/src/lnd.rs

```diff
@@ -0,0 +1,302 @@
+        let mut client = self.client.clone();
+
+        let info = client
+            .lightning()
+            .get_info(GetInfoRequest {})
+            .await
+            .expect("failed to get info")
+            .into_inner();
+
+        let pub_key: PublicKey = info.identity_pubkey.parse().expect("invalid pubkey");
+        info!("fetched pubkey {:?}", pub_key);
+
+        Ok(GetPubKeyResponse {
+            pub_key: pub_key.serialize().to_vec(),
+        })
+    }
+
+    async fn routehints(&self) -> crate::Result<GetRouteHintsResponse> {
+        Ok(GetRouteHintsResponse {
+            route_hints: vec![RouteHint { hops: vec![] }],
+        })
+    }
+
+    async fn pay(&self, invoice: PayInvoiceRequest) -> crate::Result<PayInvoiceResponse> {
+        let mut client = self.client.clone();
+
+        let send_response = client
+            .lightning()
+            .send_payment_sync(SendRequest {
+                payment_request: invoice.invoice.to_string(),
+                ..Default::default()
+            })
+            .await
+            .map_err(|e| anyhow::anyhow!(format!("LND error: {e:?}")))?
+            .into_inner();
+        info!("send response {:?}", send_response);
+
+        if send_response.payment_preimage.is_empty() {
+            return Err(GatewayError::LnRpcError(tonic::Status::new(
+                tonic::Code::Internal,
+                "LND did not return a preimage",
+            )));
+        };
+
+        Ok(PayInvoiceResponse {
+            preimage: send_response.payment_preimage,
+        })
+    }
+
+    async fn subscribe_htlcs<'a>(
+        &self,
+        subscription: SubscribeInterceptHtlcsRequest,
+    ) -> crate::Result<HtlcStream<'a>> {
+        let channel_size = 1024;
+
+        // Channel to send responses to LND after processing intercepted HTLC
+        let (lnd_tx, lnd_rx) = mpsc::channel::<ForwardHtlcInterceptResponse>(channel_size);
+
+        // Channel to send intercepted htlc to gatewayd for processing
+        let (gwd_tx, gwd_rx) =
+            mpsc::channel::<Result<SubscribeInterceptHtlcsResponse, tonic::Status>>(channel_size);
+
+        let scid = subscription.short_channel_id;
+        let mut client = self.client.clone();
+
+        let mut tg = self.task_group.clone();
+        let outcomes = self.outcomes.clone();
+
+        // Spawn an LND interceptor task
+        tg.spawn("LND HTLC Subscription", move |_handle| async move {
+            let mut htlc_stream = match client
+                .router()
+                .htlc_interceptor(ReceiverStream::new(lnd_rx))
+                .await
+                .map_err(|e| {
+                    error!("Failed to connect to lnrpc server: {:?}", e);
+                    GatewayError::Other(anyhow!("Failed to subscribe to LND htlc stream"))
+                }) {
+                Ok(stream) => stream.into_inner(),
+                Err(e) => {
+                    let _ = gwd_tx
```

Comment:
remove let _

Reply:
maybe we should love if this send fails?

Reply:
done